src/main/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseAction.java

```diff
@@ -334,18 +334,31 @@ private void trySubmitTutorial(@NotNull Project project,
     TutorialExercise exercise = tutorialViewModel.getExercise();
 
     // a submission for an IDE activity only has one file with a magic file name
-    var tutorialResultFile = FileUtilRt.createTempFile("tutorials", null);
+    var tutorialResultFile = FileUtilRt.createTempFile(Tutorial.TUTORIAL_SUBMIT_FILE_NAME, null);
     String payload = tutorial.getSubmissionPayload();
     FileUtils.writeStringToFile(tutorialResultFile, payload, StandardCharsets.UTF_8);
 
-    Map<String, Path> files = Map.of(Tutorial.TUTORIAL_SUBMIT_FILE_NAME, tutorialResultFile.toPath());
+    var submissionInfo = exercise.getSubmissionInfo();
+    String language = "en";
+
+    if (!submissionInfo.isSubmittable(language)) {
+      logger.warn("{} not submittable", exercise);
+      notifier.notify(new NotSubmittableNotification(), project);
+      return;
+    }
+    var submittableFiles = submissionInfo.getFiles(language);
+    if (submittableFiles.size() != 1) {
+      logger.warn("Tutorial doesn't have one submittable file: {}", submittableFiles);
```

Comment:
From what I understand, passing arguments to the logger converts them to strings using `toString` methods. Calling `toString` on a List doesn't produce a list of elements, instead it produces some garbage text.

I suggest doing something like
`logger.warn("Tutorial {} doesn't have one submittable file", exercise);`
since knowing *which* tutorial is faulty is much more important than what files are actually to be submitted (we can check the latter ourselves)